blingfire/src/lib.rs

```diff
@@ -1,100 +1,71 @@
 mod errors;
 
 use blingfire_sys::{TextToSentences as text_to_sentences_ffi, TextToWords as text_to_words_ffi};
-use failchain::{bail, ResultExt};
-use std::{convert::TryInto, mem};
+use failchain::ensure;
+use std::convert::TryInto;
+use std::i32;
+use std::os::raw::{c_char, c_int};
 
 pub use crate::errors::{ErrorKind, Result};
 
 #[inline]
 pub fn text_to_words(source: &str, destination: &mut String) -> Result<()> {
-    tokenize(
-        |raw_source, raw_source_len, raw_destination, raw_destination_capacity| unsafe {
-            text_to_words_ffi(
-                raw_source,
-                raw_source_len,
-                raw_destination,
-                raw_destination_capacity,
-            )
-        },
-        source,
-        destination,
-    )
+    tokenize(text_to_words_ffi, source, destination)
 }
 
 #[inline]
 pub fn text_to_sentences(source: &str, destination: &mut String) -> Result<()> {
-    tokenize(
-        |raw_source, raw_source_len, raw_destination, raw_destination_capacity| unsafe {
-            text_to_sentences_ffi(
-                raw_source,
-                raw_source_len,
-                raw_destination,
-                raw_destination_capacity,
-            )
-        },
-        source,
-        destination,
-    )
+    tokenize(text_to_sentences_ffi, source, destination)
 }
 
+type Tokenizer = unsafe extern "C" fn(*const c_char, c_int, *mut c_char, c_int) -> c_int;
```

Comment:
so I think this ends up not inlining the call, no?

Reply:
as discussed, nothing to inline